### ISSUE TITLE
Can't open new buffer

### ISSUE THREAD

**Issue by otreblan** [p00f/godbolt.nvim#3]
![image](https://user-images.githubusercontent.com/39320840/142736421-167bc5d8-8f15-4118-a05a-a8d8c8396667.png)
In Arch linux, neovim 0.5.1-2.
[Neovim config](https://github.com/otreblan/dotfiles/tree/master/.config/nvim)

**Comment by dchinmay2** [comment]
I made some changes recently, can you update and try? If the issue persists, paste the output from `:messages`

**Comment by otreblan** [comment]
```
vim/shared.lua:64: plain: expected boolean, got table
stack traceback:
^Ivim/shared.lua:64: in function 'gsplit'
^Ivim/shared.lua:111: in function 'split'
^I...nvim/site/pack/packer/start/godbolt.nvim/lua/godbolt.lua:23: in function 'prepare_buf'
^I...nvim/site/pack/packer/start/godbolt.nvim/lua/godbolt.lua:59: in function <...nvim/site/pack/packer/start/godbolt.nvim/lua/godbolt.lua:48>
```

**Comment by otreblan** [comment]
Btw the last commit was 2 hours ago

**Comment by dchinmay2** [comment]
Thanks for reporting, neovim 0.5.1 uses an older implementation of vim.split which doesn't allow extra options

`:h vim.split()` in neovim 0.5.1:
```
split({s}, {sep}, {plain})                                       *vim.split()*
                Splits a string at each instance of a separator.

                Examples: >
                 split(":aa::b:", ":")     --> {'','aa','','b',''}
                 split("axaby", "ab?")     --> {'','x','y'}
                 split(x*yz*o, "*", true)  --> {'x','yz','o'}
<

                Parameters: ~
                    {s}      String to split
                    {sep}    Separator string or pattern
                    {plain}  If `true` use `sep` literally (passed to
                             String.find)

                Return: ~
                    List-like table of the split components.

                See also: ~
                    |vim.gsplit()|
```

in neovim 0.6:

```
split({s}, {sep}, {kwargs})                                      *vim.split()*
                Splits a string at each instance of a separator.

                Examples: >

                  split(":aa::b:", ":")     --> {'','aa','','b',''}
                  split("axaby", "ab?")     --> {'','x','y'}
                  split("x*yz*o", "*", {plain=true})  --> {'x','yz','o'}
                  split("|x|y|z|", "|", {trimempty=true}) --> {'x', 'y', 'z'}
<

                Parameters: ~
                    {s}       String to split
                    {sep}     Separator string or pattern
                    {kwargs}  Keyword arguments:
                              • plain: (boolean) If `true` use `sep` literally
                                (passed to string.find)
                              • trimempty: (boolean) If `true` remove empty
                                items from the front and back of the list

                Return: ~
                    List-like table of the split components.

                See also: ~
                    |vim.gsplit()|
```

**Comment by dchinmay2** [comment]
I'll implement splitting myself

**Comment by otreblan** [comment]
It works on neovim-nightly thanks

**Comment by dchinmay2** [comment]
> It works on neovim-nightly thanks

nice

> I'll implement splitting myself

_or i could drop support for neovim 0.5 as 0.6 is going to be released soon_ this will also let me use the faster json decoding

**Comment by dchinmay2** [comment]
Closed as support for neovim 0.5 has been dropped
